js/main.js

```diff
@@ -0,0 +1,113 @@
+var $h2GeneralSetup = document.getElementById('append-gen-joke');
+var $h2GeneralPunchline = document.getElementById('append-gen-punchline');
+
+var $h2ProgrammingSetup = document.getElementById('append-prog-joke');
+var $h2ProgrammingPunchline = document.getElementById('append-prog-punchline');
+
+function programmingJokes() {
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', 'https://official-joke-api.appspot.com/jokes/programming/random');
+  xhr.responseType = 'json';
+  xhr.addEventListener('load', function () {
+    var strJokeProg = JSON.stringify(xhr.response[0].setup);
```

Comment:
Is the stringify really necessary here? This looks like it is a string already 🤔